packages/cactus-plugin-bungee-hermes/src/main/typescript/strategy/strategy-ethereum.ts

```diff
@@ -0,0 +1,228 @@
+    this.log = LoggerProvider.getOrCreate({
+      label: StrategyEthereum.CLASS_NAME,
+      level,
+    });
+  }
+
+  public async generateLedgerStates(
+    stateIds: string[],
+    networkDetails: EthereumNetworkDetails,
+  ): Promise<Map<string, State>> {
+    this.log.debug(`Generating ledger snapshot`);
+    const config = new Configuration({
+      basePath: networkDetails.connectorApiPath,
+    });
+    const ethereumApi = new EthereumApi(config);
+
+    const ledgerStates = new Map<string, State>();
+    const assetsKey =
+      stateIds.length == 0
+        ? ((await this.getAllAssetsKey(
+            networkDetails,
+            ethereumApi,
+          )) as string[])
+        : stateIds;
+    this.log.debug("Current assets detected to capture: " + assetsKey);
+    for (const assetKey of assetsKey) {
+      const { transactions, values, blocks } = await this.getAllInfoByKey(
+        assetKey,
+        networkDetails,
+        ethereumApi,
+      );
+
+      const state = new State(assetKey, values, transactions);
+
+      const stateProof = new StateProof(
+        state.getValue(),
+        parseInt(state.getVersion()),
+        state.getId(),
+      );
+      const blocksHash: string[] = [];
+      for (const block of blocks.values()) {
+        if (blocksHash.indexOf(block.hash as string) !== -1) {
+          continue;
+        }
+        blocksHash.push(block.hash as string);
+        stateProof.addBlock({
+          blockHash: block.hash as string,
+          blockCreator: block.miner as string,
+          blockSigners: [], // Non applicable for ethereum
+        });
+      }
+      state.setStateProof([]);
+      ledgerStates.set(assetKey, state);
+    }
+    return ledgerStates;
+  }
+
+  async getAllAssetsKey(
+    networkDetails: EthereumNetworkDetails,
+    api: EthereumApi,
+  ): Promise<string | string[]> {
+    const parameters = {
+      contract: {
+        contractName: networkDetails.contractName,
+        keychainId: networkDetails.keychainId,
+      },
+      //keychainId: networkDetails.keychainId,
+      invocationType: EthContractInvocationType.Call,
+      methodName: "getAllAssetsIDs",
+      params: [],
+      signingCredential: networkDetails.signingCredential,
+    };
+    const response = await api.invokeContractV1(
+      parameters as InvokeContractV1Request,
+    );
+    if (response != undefined) {
+      return response.data.callOutput;
+    }
+
+    return "response undefined";
+  }
+
+  async getAllInfoByKey(
+    key: string,
+    networkDetails: EthereumNetworkDetails,
+    api: EthereumApi,
+  ): Promise<{
+    transactions: Transaction[];
+    values: string[];
+    blocks: Map<string, EvmBlock>;
+  }> {
+    const filter = {
+      fromBlock: "earliest",
+      toBlock: "latest",
+      address: networkDetails.contractAddress,
+      topics: [null, Web3.utils.keccak256(key)], //filter logs by asset key
+    };
+    const getLogsReq: InvokeRawWeb3EthMethodV1Request = {
```

Comment:
BTW - ethereum connector exposes JSON-RPC proxy so you can use web3 client directly instead of these endpoints (which also heavily depend on the web3js release used by the connector), see https://github.com/hyperledger/cacti/tree/main/packages/cactus-plugin-ledger-connector-ethereum#json-rpc-proxy

Reply:
Using "raw" endpoints works as well but proxy might be a better alternative in your future contributions :)

Reply:
Thank you! I will look into this.